### DISCUSSION TITLE
Doubt about nested Resources/Relationships: INVOICE > SUPLLY_DATE > PRODUCT

### DISCUSSION THREAD

**Discussion by alexworldbr** [filamentphp/filament#8604] [Help] [answered]
Hi Guys,

I have a "challenge" to craft a form resource with nested resources and relationships.

1. INVOICE (parent), that contains many SUPPLY_DATE
2. SUPPLY_DATE (child), that contains many dates supplies, with one or more PRODUCTS
3. PRODUCT (subchild), that contains product value and quantity supplied each date.

**What is the best approach or suggestion for building this form?** Preferably with relationships and without modal.

Any suggestions?

The final table would be this sample:

> Invoice 001	(#resource INVOICE)
> 
> 	Supply	2023/01/01	($resource SUPPLY by date)
> 
> 		product		        qty	sum
> 
> 		Prodcut 001 (price U$ 100)	100	10,000		(#resource PRODUCT)
> 		Product 002 (price U$ 200)	500	100,000
> 		Product 003 (price U$ 300)	800	240,000
> 
> 	Supply	2023/01/02
> 
> 		Prodcut 001 (price U$ 100)	200	20,000
> 		Product 002 (price U$ 200)	600	120,000
> 		Product 003 (price U$ 300)	900	270,000
> 
> 	Supply	2023/01/03
> 
> 		Prodcut 001 (price U$ 100)	300	30,000
> 		Product 002 (price U$ 200)	700	140,000
> 		Product 003 (price U$ 300)	400	120,000
> 
> total invoice	1,050,000

The ideal would be a form similar to a spreadsheet like this:

Invoice 

................................ |  Prod. 1 | Prod. 2 | Prod. 3 |
Supply 2023/01/01 |    100    |   200      |     300   |
Supply 2023/01/02 |    ...        
Supply 2023/01/03 |    ...       

Tks,

Alex
from Brazil

**Comment by leandrocfe** [comment, marked as answer]
Maybe you could take a look at the [demo](https://demo.filamentphp.com/), as it includes information on Orders, Order Items, Products, Comments..."

**Reply by alexworldbr** [reply]
Bom Dia Leandro,

Eu havia analisado justamente o demo do filament, mas dentro de "Order items" precisaria ter a opção de agrupar produtos por data, onde em cada data haveria os fornecimentos do dia.

Outra opção que pensei foi no resource/form INVOICE criar uma relation/table de SUPPLIES ("fornecimentos") com campos input/numeric agrupados por data para os PRODUCTS. Será que funcionaria assim?

Grato pela dica!

**Reply by leandrocfe** [reply]
Bom dia, sim você pode criar o resource de fornecimentos. O Relation Manager é uma forma de você editar o relacionamento dentro do modelo, mas você pode criar como um resource.

**Reply by alexworldbr** [reply]
Ok Leandro, vou tentar aqui com o agrupamento e te digo qualquer dúvida.
